File: agent/android/Crescent/app/src/main/java/com/displaynote/crescent/StartServiceReceiver.java
package com.displaynote.crescent;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class StartServiceReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Util.startJob(context);
    }
}

File: agent/android/Crescent/app/src/main/java/com/displaynote/crescent/StartServiceReceiver.kt
package com.displaynote.crescent

import android.annotation.SuppressLint
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import com.displaynote.crescent.Util.startJob

class StartServiceReceiver : BroadcastReceiver() {
    @SuppressLint("UnsafeProtectedBroadcastReceiver")
    override fun onReceive(context: Context, intent: Intent) {
        Log.d(TAG, "onReceive")
        IoTSystem.init(context)
        Hardware.init(context)
        startJob(context)
    }

    companion object {
        const val TAG = "StartServiceReceiver"
    }
}